pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+                                    LocalOnly=True,
+                                    IncludeQualifiers=False)
+            except CIMError as ce:
+                if ce.args[0] == CIM_ERR_NOT_FOUND:
+                    ce.args = (CIM_ERR_INVALID_SUPERCLASS, new_class.superclass)
+                    raise
+                else:
+                    raise
+
+        if new_class.classname in self.classes[namespace]:
+            raise CIMError(CIM_ERR_ALREADY_EXISTS, new_class.classname)
+
+        self.classes[namespace][new_class.classname] = new_class
+
+    def _fake_deleteclass(self, namespace, **params):
+        """
+        Implements a mock server reactor for
+        :meth:`~pywbem.WBEMConnection.DeleteClass`
+
+        Delete a class in the class repository if it exists.
+        Emulates WBEMConnection.DeleteClass(...))
+
+        This is simplistic in that it ignores issues like existing
+        subclasses and existence of instances.
+
+        Nothing is returned.
+
+        Exceptions:
+            CIMError CIM_ERR_NOT_FOUND
+        """
+        self._validate_classes_namespace(namespace)
+
+        cname = params['ClassName'].classname
+
+        try:
+            del self.classes[namespace][cname]
+        except KeyError as ke:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Cannot Delete %s. Not in '
+                           'repository %s' % (cname, ke))
+
+    ##########################################################
+    #
+    #              Faked Qualifier methods
+    #
+    ###########################################################
+
+    def _fake_enumeratequalifiers(self, namespace, **params):
+        # pylint: disable=unused-argument
+        """
+        Imlements a mock server reactor for
+        :meth:`~pywbem.WBEMConnection.EnumerateQualifiers`
+
+        Enumerates the qualifier declarations in the local repository of this
+        namespace.
+
+        Returns: TODO
+
+        Exceptions:
+            CIMError
+
+        """
+        try:
+            qualifiers = list(self.qualifiers[namespace].values())
+        except KeyError:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
+        return self._make_tuple(qualifiers)
+
+    def _fake_getqualifier(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`pywbem.WBEMConnection.GetQualifier`.
+
+        Retrieves a qualifier declaration from the local repository of this
+        namespace.
+
+        Return:
+          Returns a tuple representing the _imethodcall return for this
+          method where the data is a QualifierDeclaration
+
+        Exceptions:
+            CIMError CIM_ERR_INVALID_NAMESPACE, CIMError(CIM_ERR_NOT_FOUND
+        """
+        if not self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
```

Comment:
We should use the generic namespace validation function I suggested in another comment.

Reply:
Done